packages/react/src/components/TextField/TextField.tsx

```diff
@@ -16,29 +16,108 @@
+import ListItem from '../ListItem';
+import ListItemIcon from '../ListItemIcon';
+import ListItemText from '../ListItemText';
+import Tooltip from '../Tooltip';
 import './text-field.scss';
 
-export type TextFieldProps = MuiTextFieldProps;
+export type TextFieldProps = {
+  /**
+   * Criteria for user input.
+   */
+  criteria?: string[];
+} & MuiTextFieldProps;
 
 const COMPONENT_NAME: string = 'TextField';
 
 const TextField: FC<TextFieldProps> & WithWrapperProps = (props: TextFieldProps): ReactElement => {
-  const {className, label, InputLabelProps, id, ...rest} = props;
+  const {className, criteria, InputLabelProps, id, label, type, ...rest} = props;
 
   const classes: string = clsx('oxygen-text-field', className);
 
+  const [open, setOpen] = useState<boolean>(false);
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClick = (): void => {
+    if (type === 'password' && criteria?.length > 0) {
+      setOpen(true);
+    }
+  };
+
+  const handleClose = (): void => {
+    if (open) {
+      setOpen(false);
+    }
+  };
+
+  const handleClickShowPassword = (): void => setShowPassword((show: boolean) => !show);
+
+  const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>): void => {
+    event.preventDefault();
+  };
+
+  const tooltipContent = (): ReactNode => (
+    <List>
+      {criteria?.map((criterion: string) => (
+        <ListItem disablePadding>
+          <ListItemIcon>
+            <DoubleCircleIcon />
+          </ListItemIcon>
+          <ListItemText primary={criterion} />
+        </ListItem>
+      ))}
+    </List>
+  );
+
   return (
     <div className={classes}>
-      <InputLabel htmlFor="id" shrink className="oxygen-text-field__label" {...InputLabelProps}>
+      <InputLabel htmlFor={id} aria-describedby={id} {...InputLabelProps}>
         {label}
       </InputLabel>
-      <MuiTextField className="oxygen-text-field__input" id={id} {...rest} />
+      <Tooltip
+        arrow
+        placement="right-end"
+        describeChild
+        open={open}
+        title={tooltipContent()}
+        classes={{arrow: 'oxygen-text-field-tooltip-arrow', tooltip: 'oxygen-text-field-tooltip'}}
+      >
+        <MuiTextField
+          id={id}
+          type={type === 'password' && (showPassword ? 'text' : 'password')}
+          InputProps={
+            type === 'password' && {
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton
+                    aria-label="toggle password visibility"
+                    onClick={handleClickShowPassword}
+                    onMouseDown={handleMouseDownPassword}
+                    edge="end"
+                  >
+                    {showPassword ? <VisibilityOffIcon /> : <VisibilityIcon />}
+                  </IconButton>
+                </InputAdornment>
+              ),
+            }
+          }
+          onClick={handleClick}
+          onBlurCapture={handleClose}
```

Comment:
Since TextFiled is generic, callbacks should be handled generically.
```
const handleBlurCapture = () => {
    if (type === password) {
        // do password input related logic
    }
}


    onBlurCapture={handleBlurCapture}
```

Reply:
Fixed with https://github.com/wso2/oxygen-ui/pull/38/commits/f2ee9ffa587d3b161c78343d67540bd081438984